cocos2d/core/layers/CCLayer.js

```diff
@@ -322,21 +322,38 @@ cc.LayerGradient = cc.LayerColor.extend(/** @lends cc.LayerGradient# */{
     _alongVector: null,
     _compressedInterpolation: false,
     _className: "LayerGradient",
+    _colorStops: [],
 
     /**
      * Constructor of cc.LayerGradient
      * @param {cc.Color} start
      * @param {cc.Color} end
      * @param {cc.Point} [v=cc.p(0, -1)]
-     */
-    ctor: function (start, end, v) {
+     * @param {Array|Null} stops
+     *
+     * @example Using ColorStops argument:
+     * //startColor & endColor are for default and backward compatibility
+     * var layerGradient = new cc.LayerGradient(cc.color.RED, new cc.Color(255,0,0,0), cc.p(0, -1),
+     *                                          [{p:0, color: cc.color.RED},
+     *                                           {p:.5, color: new cc.Color(0,0,0,0)},
+     *                                           {p:1, color: cc.color.RED}]);
+     * //where p = A value between 0.0 and 1.0 that represents the position between start and end in a gradient
+     *
+     */
+    ctor: function (start, end, v, stops) {
```

Comment:
I think start and end color should to add the stops, the codes create LayerGradient should to like these:
```javascript
        var layerGradient = new cc.LayerGradient(cc.Color.RED, cc.Color.GREEN, cc.p(0,-1),
            [//{p:0, color: cc.color.RED}, 
             {p:0.5, color: new cc.Color(0,255,255,128)}
             //{p:1, color: cc.color.GREEN}
            ]);
```
or
```javascript
        var layerGradient = new cc.LayerGradient(
            [{p:0, color: cc.color.RED},
                {p:0.5, color: new cc.Color(0,255,255,128)},
                {p:1, color: cc.color.GREEN}
            ], cc.p(0,-1));
```